src/main/java/com/sonyericsson/jenkins/plugins/bfa/db/MongoDBKnowledgeBase.java

```diff
@@ -323,12 +322,14 @@ public FailureCause removeCause(String id) {
      * @see MongoDBKnowledgeBase#addCause(FailureCause)
      */
     public FailureCause addCause(FailureCause cause, boolean doUpdate) {
-        WriteResult<FailureCause, String> result = getJacksonCollection().insert(cause);
+        getJacksonCollection().insert(cause);
         if (doUpdate) {
             initCache();
             cache.updateCache();
         }
-        return result.getSavedObject();
+
+        final FailureCause modifiedFailureCause = getJacksonCollection().find(eq("name", cause.getName())).first();
```

Comment:
wouldn't the `id` be better here because it's unique? Couldn't name have multiple entries?

Reply:
Yes, you are correct, name is not unique in any way. Currently we never use the FailureCause we get back anyways, but if we started using it for some reason then weird bugs could appear, so good find!